### DISCUSSION TITLE
New release or allow to be installed from github

### DISCUSSION THREAD

**Discussion by pjdowner** [CoderLine/alphaTab#1453] [Q&A] [answered]
I need to use the develop branch as there's some bug fixes that I need, it seems to have been a long time since there's been a release so I'm wondering if there might be one soon?

Regardless I have found that the easiest current way for me to use alphatab is to just add it to my package.json, because you can now (I understand it's a new feature) install direct from git repositories I've forked alphatab and removed the dist directory from the gitignore and added the dist/* files. This allows me to use a current version and specify a commit that works well from my tests. I'm wondering if this is something upstream might adopt or do you have any other suggestions for easily following the develop branch?

https://docs.npmjs.com/cli/v9/configuring-npm/package-json?v=true#git-urls-as-dependencies

**Comment by Danielku15** [comment, marked as answer]
Hey there. 

It was indeed a long journey for 1.3 until now. A lot of problems arised in the development of the features I wanted to bring in this release and most things were out of our control. While Kotlin advertises itself as cross compilation platform it had many restrictions and missing features which tend to break over and over on the different releases. I managed to target the native Android platform but had to drop iOS due to too many limitations.

I changed the focus of this release lately to improve the Web Platform support with bundlers and also hit again many walls with them also not supporting many required features. But also there I lately managed to work around most things and most things which should be in the release are not there. But there is still a bit of documentation and testing to be done before 1.3 will be released.  

Generally the release progress and contents can always be followed in GitHub Projects: https://github.com/orgs/CoderLine/projects/13

>  I'm wondering if this is something upstream might adopt or do you have any other suggestions for easily following the develop branch?

While I generally not recommend to always target the latest develop (as it can break) we have quasi-nightly builds available for all packages. They are released as pre-release versions to all package managers and can be consumed from there: 

- https://www.npmjs.com/package/@coderline/alphatab?activeTab=versions
- https://www.nuget.org/packages/AlphaTab/1.3.0-alpha.1008#versions-body-tab
- https://s01.oss.sonatype.org/content/repositories/snapshots/net/alphatab/

No need for any workarounds or tricks beside pulling in the pre-release versions 😉 But I recommend to test things through and maybe stick to a specific version you know is stable for your needs.

**Comment by pjdowner** [comment]
Oh that's awesome, thanks for such a detailed and quick answer. I'll take a look at the pre-releases, sorry if I missed that in the documentation!
